arangod/RocksDBEngine/RocksDBChecksumEnv.h

```diff
@@ -0,0 +1,101 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <openssl/evp.h>
+#include <rocksdb/env.h>
+
+#include "Basics/Mutex.h"
+
+namespace arangodb::checksum {
+
+class ChecksumCalculator {
+ public:
+  ChecksumCalculator();
+  void computeFinalChecksum();
+  void updateIncrementalChecksum(char const* buffer, size_t n);
+  void updateEVPWithContent(char const* buffer, size_t n);
+  [[nodiscard]] std::string getChecksum() { return _checksum; }
+  ~ChecksumCalculator();
+
+ private:
+  EVP_MD_CTX* _context;
+  std::string _checksum;
+};
+
+class ChecksumHelper {
+ public:
+  explicit ChecksumHelper(std::string const& rootPath) : _rootPath{rootPath} {}
+  [[nodiscard]] static bool isFileNameSst(std::string const& fileName);
+  bool writeShaFile(std::string const& fileName, std::string const& checksum);
+  [[nodiscard]] std::string buildShaFileNameFromSst(
```

Comment:
AFAIK the method does not refer to any instance variables, so it could be made static.
```suggestion
  [[nodiscard]] static std::string buildShaFileNameFromSst(
```